aten/src/ATen/core/function_schema.h

```diff
@@ -373,6 +451,61 @@ struct FunctionSchema {
 
     return is_mutable(*index);
   }
+  bool areAliasing(const SchemaArgument& lhs, const SchemaArgument& rhs) const {
+    TORCH_INTERNAL_ASSERT(
+        (lhs.index < getCorrectList(lhs.type).size()),
+        "Invalid index for schema.");
+    TORCH_INTERNAL_ASSERT(
+        (rhs.index < getCorrectList(rhs.type).size()),
+        "Invalid index for schema.");
+
+    const Argument lhsArg = getCorrectList(lhs.type)[lhs.index];
+    const Argument rhsArg = getCorrectList(rhs.type)[rhs.index];
+    if (lhsArg.alias_info() && rhsArg.alias_info()) {
+      for (const auto& lhsSet : lhsArg.alias_info()->afterSets()) {
+        for (const auto& rhsSet : rhsArg.alias_info()->afterSets()) {
+          if (lhsSet == rhsSet) {
+            return true;
+          }
+        }
+      }
+    }
+    return false;
+  }
+  bool areAliasing(const SchemaArgument& lhs, const SchemaArgument& rhs, bool check_additional) const {
```

Comment:
Please document what `check_additional` means.

Also, is there a need to expose two versions of this function as public?